src/main/java/no/nav/fo/veilarbportefolje/consumer/DialogDataFeedHandler.java

```diff
@@ -38,21 +41,20 @@ public DialogDataFeedHandler(BrukerRepository brukerRepository,
     @Transactional
     public void call(String lastEntry, List<DialogDataFraFeed> data) {
 
-        try {
-            timed("feed.dialog.objekt",
-                    () -> {
-                        data.forEach(info -> {
-                            dialogFeedRepository.oppdaterDialogInfoForBruker(info);
-                            indekseringService.indekserAsynkront(AktoerId.of(info.getAktorId()));
-                        });
-                    },
-                    (timer, hasFailed) -> timer.addTagToReport("antall", Integer.toString(data.size()))
-            );
-            brukerRepository.updateMetadata(DIALOGAKTOR_SIST_OPPDATERT, Date.from(ZonedDateTime.parse(lastEntry).toInstant()));
-        } catch(Exception e) {
-            log.error("Feil ved behandling av dialogdata fra feed for liste med brukere.", e);
-        }
-
-        MetricsFactory.createEvent("datamotattfrafeed").report();
+        List<Runnable> jobs = data.stream()
+                .map(this::toRunnable)
+                .collect(toList());
+
+        CompletableFuture future = async(jobs);
+        await(future);
```

Comment:
Slo på stortromma her og la til en veldig rudimentær async/await-mekanisme (som vi kanskje kunne lagt i fellesbibliotekene etterhvert). Det er bare et tynt lag over CompletableFutures, men langt mer lesbart vil jeg tørre å påstå.

Her gir man bare `async()` en liste over Runnables hvor samtlige jobber i listen blir kjørt, uavhengig av om noen av de kaster exception. Den kortslutter altså ikke siden exceptions blir kastes på en egen tråd. `await()` vil blokke eksekveringen helt til alle jobbene er ferdig kjørt. Akkurat nå kjøres jobbene på én og samme tråd, men kunne potensielt gitt den flere tråder for litt parallelitet. 

Etter hvert kunne vi utvidet `async()`, hvor den tar inn Suppliers også, og dermed kan `await()` returnere resultatet av jobben(e).

Tanker @gjengeda?

Reply:
@alexander-svendsen

Reply:
Nå vil det bli laget en Runnable av hver av elementene fra feeden, som så vil bli oppdatert uavhengig av hverandre, ikke sant? Og da bryr den seg heller ikke om exceptions, dvs. at hvis det er 10 elementer i listen vil alle 10 bli forsøkt prosessert selv om én eller flere av dem feiler? Stemmer det at vi nå også _alltid_ vil oppdatere metadata uavhengig av om noe feiler eller ikke?
I så fall får vi nå en litt annen oppførsel enn tidligere, siden vi da ville avbryte (og ikke oppdatere metadata) om et av elementene feilet. Da ville feeden bli lest inn på nytt fra samme sted. Vi ville dermed være sikre på alle innleste elementer faktisk ble prosessert ok, men samtidig ville innlesing av feeden stoppe opp dersom et feed-element konsekvent feilet. Hvis jeg forstår den nye koden riktig, vil et element som feiler nå bare bli forkastet, mens metadata fortsatt blir oppdatert og innlesing går videre? Eller vil feilhåndteringsmekanismen i await kaste en exception som medfører at metadata ikke blir oppdatert?